src/qt/sendcoinsentry.cpp

```diff
@@ -78,7 +78,14 @@ void SendCoinsEntry::on_addressBookButton_clicked()
 
 void SendCoinsEntry::on_payTo_textChanged(const QString &address)
 {
-    updateLabel(address);
+    SendCoinsRecipient rcp;
+    if (GUIUtil::parseBitcoinURI(address, &rcp)) {
+        setValue(rcp);
+    }
+
+    if (rcp.label.isEmpty()) {
```

Comment:
This way the label of the addressbook will become replaced if there is a label in the URI. 
 
Do we wan't to keep this behaviour or rather remove this check and just replace the URI label with the addressbook label?

Reply:
This works but only when pasting into `payTo`, had to patch it c03af67ea6 to ensure the same behaviour for "Open URI"

Reply:
good catch, applied this along with another small tweak to avoid a redundant invocation of `updateLabel` , see b97aed1